lib/Optimizer/Support/InternalNames.cpp

```diff
@@ -186,6 +189,12 @@ fir::NameUniquer::doVariable(llvm::ArrayRef<llvm::StringRef> modules,
   return result.append(toLower(name));
 }
 
+llvm::StringRef fir::NameUniquer::doProgramEntry() {
+  if (mainEntryName.size())
+    return mainEntryName;
```

Comment:
Should this be `_QQ` + mainEntryName so that we retain the info that this is the main program ?

Reply:
My thought is that if someone wants to override the name, then we don't want to get in the way at all.